src/tooltip.tsx

```diff
@@ -135,10 +142,64 @@ const Tooltip: React.FC<Props> = ({
         height: 0,
     });
 
-    const getPosition = (position: Position = defaultPositionDesktop) =>
-        isTabletOrSmaller && (position === 'left' || position === 'right') ? defaultPositionMobile : position;
+    const [containerPosition, setContainerPosition] = React.useState({});
+
+    const getPosition = (position: Position | undefined) =>
+        isTabletOrSmaller ? position || defaultPositionMobile : position || defaultPositionDesktop;
+
+    const noRightSpace = (position: Position) => {
```

Comment:
why do you pass `position` to this function? it doesn't make sense, this function should just return a boolean, and the postition isn't relevant.

Also, try to avoid negations in function names. I'd call this function `hasRightSpace`

Reply:
the reason why it had position was that the tooltipRef ended up complaining if it could not return anything because it could be null, to work around this and not need to pass the position, I changed it to return as undefined